pkg/hubble/parser/threefour/policy.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2019 Authors of Hubble
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package threefour
+
+import (
+	"strconv"
+
+	"github.com/cilium/cilium/api/v1/flow"
+	"github.com/cilium/cilium/pkg/k8s"
+	slim_networkingv1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/apis/networking/v1"
+	"github.com/cilium/cilium/pkg/labels"
+	monitorAPI "github.com/cilium/cilium/pkg/monitor/api"
+	"github.com/cilium/cilium/pkg/policy/api"
+)
+
+func flowMatchesNetworkPolicy(fl *flow.Flow, policy *slim_networkingv1.NetworkPolicy) bool {
```

Comment:
Did you consider having this interact directly with either the policy repository or the selectorpolicy layers? Integrating there would allow all policy types to be supported at once, and would ideally reuse the same logic for both generating the entries (for enforcement) and for evaluating which policies caused this match to occur (for visibility here).

Reply:
I will investigate this.